validators_test.go

```diff
@@ -119,3 +119,38 @@ func TestValidators__validateSimpleTime(t *testing.T) {
 		}
 	}
 }
+
+// Ensure that ASCII characters from 0x20 to 0x7E are considered alphanumeric.
+func TestValidators__isAlphanumeric(t *testing.T) {
```

Comment:
Nice refactor. Also, I would recommend combining these two tests into one using table-driven tests since they share the same test logic, here's an example:

```go
func TestValidators__isAlphanumeric(t *testing.T) {
	v := validator{}

	tests := []struct {
		name      string
		checkFunc func(string) error
		shouldErr func(i int) bool
	}{
		// Ensure that ASCII characters from 0x20 to 0x7E are considered alphanumeric.
		{
			name:      "is alphanumeric",
			checkFunc: v.isAlphanumeric,
			shouldErr: func(i int) bool {
				return i < 0x20 || i > 0x7E
			},
		},
		// Ensure that ASCII characters from 0x20 to 0x60 and 0x7B to 0x7E are considered upper case alphanumeric.
		{
			name:      "is upper alphanumeric",
			checkFunc: v.isUpperAlphanumeric,
			shouldErr: func(i int) bool {
				return i < 0x20 || i > 0x7E || (i > 0x60 && i < 0x7B)
			},
		},
	}

	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			for i := 0; i < 255; i++ {
				chr := string(rune(i))
				err := tt.checkFunc(chr)

				shouldError := tt.shouldErr(i)

				if shouldError && err == nil {
					t.Errorf("expected rune %x (%s) to be non-alphanumeric", i, chr)
				} else if !shouldError && err != nil {
					t.Errorf("expected rune %x (%s) to be alphanumeric: %v", i, chr, err)
				}
			}
		})
	}
}
```

Reply:
Adopted this.